sdk/eventgrid/Azure.Messaging.EventGrid/src/Customization/EventGridPublisherClient.cs

```diff
@@ -119,10 +119,20 @@ private async Task<Response> PublishEventsInternal(IEnumerable<EventGridEvent> e
                     // Individual events cannot be null
                     Argument.AssertNotNull(egEvent, nameof(egEvent));
 
-                    MemoryStream stream = new MemoryStream();
-                    _dataSerializer.Serialize(stream, egEvent.Data, egEvent.Data.GetType(), cancellationToken);
-                    stream.Position = 0;
-                    JsonDocument data = JsonDocument.Parse(stream);
+                    JsonDocument data;
+                    if (egEvent.Data is BinaryData binaryEventData)
```

Comment:
Can we add a test that covers this? The generated request should have Json of BinaryData, but rather the thing that BinaryData was wrapping.

Reply:
Turns out BinaryData can't be serialized anyway.